postprocessing/science/get_distances_to_rupture.jl

```diff
@@ -0,0 +1,195 @@
+    out = copy(transpose(convert(Matrix{Float32}, xyz)))
+    return out
+end
+
+
+function get_ASl(xdmf, event)
+    timesteps = timesteps_of(xdmf)
+    if event == 0
+        ASl = data_of(xdmf, size(timesteps)[1], "ASl")
+    elseif event == 1
+        ASl = data_of(xdmf, Int(round(size(timesteps)[1] / 2) - 1), "ASl")
+    elseif event == 2
+        ASl = data_of(xdmf, size(timesteps)[1], "ASl") .- data_of(xdmf, Int(round(size(timesteps)[1] / 2) - 1), "ASl")
+    else
+        println("Only 0, 1, 2 are allowed for the event argument.")
+        ccall(:jl_exit, Cvoid, (Int32,), 86) # Exit code 86 to the O/S
+    end
+    return convert(Vector{Float32}, ASl)
+end
+
+
+function get_distances_to_rupture(ASl::Vector{Float32}, xyz_fault::Matrix{Float32}, xyz_receiver::Matrix{Float32}, 
+        slip_threshold::Float32)
+    
+    distances = Vector{Float32}(undef, size(xyz_receiver)[1])
+    rupturing_fault = xyz_fault[ASl .> slip_threshold,:]
+    if rank == 0
+        println("Using $(Threads.nthreads()) threads per MPI rank...")
+    end
+    Threads.@threads for i in 1:size(xyz_receiver)[1]
+        distances[i] = minimum(sqrt.(sum(((rupturing_fault .- reshape(xyz_receiver[i,:], 1,3)).^2), dims=2)))
+    end
+    return distances
+end
+
+
+if use_gpu
+    using CUDA
+    function get_distances_to_rupture(ASl::CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}, 
+                        xyz_fault::CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, 
+                        xyz_receiver::CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, 
+                        slip_threshold::Float32)
+        # gpu version of the function (employing multiple dispatch)
+
+        rupturing_fault = xyz_fault[ASl .> slip_threshold,:]
+        distances = Vector{Float32}(undef, size(xyz_receiver)[1])
+        for i in 1:size(xyz_receiver)[1]
+            distances[i] = minimum(sqrt.(sum(((rupturing_fault .- reshape(xyz_receiver[i,:], 1,3)).^2), dims=2)))
+        end
+        return distances
+    end
+end
+
+
+function get_chunk_sizes_and_offsets(N::Integer, n::Integer)
+    q,r = divrem(N, n)
+    lengths = [i <= r ? q+1 : q for i = 1:n]
+    offsets = [i == 1 ? 0 : sum(lengths[1:i-1]) for i = 1:n]
+    return lengths, offsets
+end
+
+
+if noMPI
+    rank = 0
+    nRanks = 1
+else
+    using MPI
+    
+    MPI.Init()
+    comm = MPI.COMM_WORLD
+    rank = MPI.Comm_rank(comm)
+    nRanks = MPI.Comm_size(comm)
+end
+
+
+timeOutput = @timed begin
+    
+    xdmf = XDMFFile(filename)
+    
+    if split(filename, '-')[end] == "surface.xdmf"
+        filename2 = filename[1:end-12]*"fault.xdmf"
+    elseif split(filename, '-')[end] == "volume.xdmf"
+        filename2 = filename[1:end-11]*"fault.xdmf"
+    elseif split(filename, '.')[end] == "xdmf"
+        filename2 = filename[1:end-5]*"-fault.xdmf"
+    else
+        println("Filename deviates from standard SeisSol output, can not find ...-fault.xdmf")
```

Comment:
it seems the error message is misleading.
>Filename deviates from standard SeisSol output, because not ending by ...

Reply:
Changed.